main_test.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"testing"
+	"time"
+)
+
+func TestDateParamToEpoch(t *testing.T) {
+
+	timeNow = func() time.Time {
+		//16 Aug 1994 15:30
+		return time.Date(1994, time.August, 16, 15, 30, 0, 100, defaultTimeZone)
+	}
+
+	const shortForm = "15:04 2006-Jan-02"
+
+	var tests = []struct {
+		input  string
+		output string
+	}{
+		{"midnight", "00:00 1994-Aug-17"},
+		{"noon", "12:00 1994-Aug-16"},
+		{"teatime", "16:00 1994-Aug-16"},
+
+		{"noon 08/16/94", "12:00 1994-Aug-16"},
+		{"midnight 20060816", "00:00 2006-Aug-16"},
+
+		{"15:04 19940816", "15:04 1994-Aug-16"},
+		{"-1day", "15:30 1994-Aug-15"},
+	}
+
+	for _, tt := range tests {
+		parsedTime := dateParamToEpoch(tt.input, int64(0))
+		ts, err := time.ParseInLocation(shortForm, tt.output, defaultTimeZone)
+		if err == nil {
```

Comment:
If we fail to parse out expected output time from our test, that's a programmer error.  I'm happy to have a `panic()`, here, if you want to be more friendly:
```go
if err != nil {
    t.Fatalf("error parsing time: %q: %v", tt.output, err), 
}
```

This follow the standard "straight-line" code layout for Go: https://medium.com/mat-ryer/line-of-sight-in-code-186dd7cdea88